src/frontend/sharedComponents/FormattedData.js

```diff
@@ -0,0 +1,154 @@
+  return <>{value}</>;
+};
+
+// Render a field value as a string. If the value is an array, convert to string
+// and join with `, `. If the field is a select_one or select_multiple field,
+// then use `field.option.label` to display the value, if a label is defined.
+// Translate the field value if a translation is defined.
+//
+// TODO: Consider an API like
+// https://formatjs.io/docs/react-intl/components#formatteddateparts to enable
+// formatting of individual items in an array value.
+export const FormattedFieldValue = ({
+  value,
+  field,
+}: {|
+  value: any,
+  field: Field,
+|}) => {
+  const { formatMessage: t } = useIntl();
+  // Select multiple answers are an array, so we join them with commas
+  const formattedValue = (Array.isArray(value) ? value : [value])
+    // Filter any undefined values or empty strings (an empty string can come
+    // from a user deleting an answer) TODO: Values that are just spaces
+    .filter(value => typeof value !== "undefined" && value !== "")
+    .map(value =>
+      t({
+        id: `fields.${field.id}.options.${JSON.stringify(value)}`,
+        defaultMessage: getValueLabel(value, field),
+      }).trim()
+    )
+    .join(", ");
+  // This will return a noAnswer string if formattedValue is undefined or an
+  // empty string
+  return <>{formattedValue || t(m.noAnswer)}</>;
+};
+
+// Format the created_at date of an observation as either a datetime, or a
+// relative datetime (e.g. "3 minutes ago")
+export const FormattedObservationDate = ({
+  observation,
+  variant,
+}: {
+  observation: Observation,
+  // 'relative' = relative date format e.g. "3 minutes ago"
+  // for other formats see formats.date
+  variant: "relative" | $Keys<typeof formats.date>,
+}) => {
+  const createdAtDate = new Date(observation.created_at);
+  switch (variant) {
+    case "relative":
+      return <DateDistance date={createdAtDate} />;
+    default:
+      return <FormattedDate value={createdAtDate} format={variant} />;
+  }
+};
+
+// Format the translated preset name, with a fallback to "Observation" if no
+// preset is defined
+export const FormattedPresetName = ({
+  preset,
+}: {|
+  preset: Preset | PresetWithFields | void,
+|}) => {
+  const { formatMessage: t } = useIntl();
+  const name = preset
+    ? t({ id: `presets.${preset.id}.name`, defaultMessage: preset.name })
+    : t(m.observation);
+
+  return <>{name}</>;
+};
+
+// TODO: Better hangling of boolean and null values (we don't create these
+// anywhere yet)
+function getValueLabel(
+  value: null | boolean | number | string,
+  field: Field
+): string {
+  if (field.type === "select_one" || field.type === "select_multiple") {
+    // Look up label from field options. This is not necessary for presets
+    // created with mapeo-settings-builder@^3.1.0, which will have these options
+    // in the translation file, but is needed for older versions of presets
+    const options = convertSelectOptionsToLabeled(field.options);
```

Comment:
is there ever a time someone would use field.options in a Mapeo React component without having it already be parsed by the 'convertSelectOptionsToLabeled' function?

Reply:
I don't think so, so perhaps we can transform options on the server or at the client api level, and treat `Array<string>` select options as a deprecated option that we change before the front-end receives it.